### PR TITLE
Support non-tensorflow backends in KerasCV's preprocessing layers

### PR DESCRIPTION
**Pull request by tirthasheshpatel**
# What does this PR do?

This PR adds support for non-tensorflow backends in KerasCV's preprocessing layers. When not working with TensorFlow specific data structures (raggeds and sparse tensors), the preprocessing layers now accept and return backend native tensors.

Note that all the computations still happen in TensorFlow and it would require a major refactor to change that.

For the TensorFlow backend, the changes in this PR should be a no-op i.e. the behaviour should remain completely unchanges for the tensorflow backend.

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: keras_cv/backend/config.py [hunk 1]
```diff
@@ -21,7 +21,8 @@ def detect_if_tensorflow_uses_keras_3():
         # Note that only recent versions of keras have a `version()` function.
         if hasattr(keras, "version") and keras.version().startswith("3."):
             return True
-    except:
+    except Exception as e:
+        raise e
```
Comment (tirthasheshpatel) [inline keras_cv/backend/config.py:25]: Unrelated change, will revert in the next batch of commits

**Comment by divyashreepathihalli** [comment]
Thanks for the PR Tirth! LGTM! Please rebase with master as some of the errors have been resolved in master.

File: keras_cv/backend/config.py (hunk 1 above)
Comment (divyashreepathihalli) [reply keras_cv/backend/config.py:25]: okay

**Review by divyashreepathihalli — APPROVED** [review]

**Comment by divyashreepathihalli** [comment]
LGTM! Thanks :)!
